### PR TITLE
docs(api.md): use async/await instead of then

### LINKED ISSUE #5082: Update docs/api.md to use async/await instead of Promise.prototype.then
Now that we've dropped support for Node.js v6, let's update the remaining code examples on https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md to use `async`/`await`. It makes them simpler and easier to follow.

Help welcome!

### PR DESCRIPTION
**Pull request by randombenj**
fixes #5082

### CODE REVIEW

**Comment by randombenj** [comment]
@googlebot I signed it!

File: docs/api.md [hunk 1]
```diff
@@ -1474,16 +1483,18 @@ An example of adding a `window.readfile` function into the page:
...
   await page.exposeFunction('readfile', async filePath => {
     return new Promise((resolve, reject) => {
       fs.readFile(filePath, 'utf8', (err, text) => {
-        if (err)
+        if (err) {
           reject(err);
-        else
+        } else {
           resolve(text);
+        }
```
Comment (mathiasbynens) [inline docs/api.md]: This now no longer matches the current Puppeteer coding style.

File: docs/api.md [hunk 2]
```diff
@@ -2560,18 +2577,20 @@ An example of dumping frame tree:
...
+  const browser = await puppeteer.launch();
   const page = await browser.newPage();
   await page.goto('https://www.google.com/chrome/browser/canary.html');
   dumpFrameTree(page.mainFrame(), '');
   await browser.close();
 
   function dumpFrameTree(frame, indent) {
     console.log(indent + frame.url());
-    for (let child of frame.childFrames())
+    for (let child of frame.childFrames()) {
```
Comment (mathiasbynens) [inline docs/api.md]:
```suggestion
    for (const child of frame.childFrames()) {
```

**Review by mathiasbynens — APPROVED** [review]

**Review by mathiasbynens — APPROVED** [review]

**Comment by mathiasbynens** [comment]
Thanks!

**Pull request merged**